### ISSUE TITLE
Output Value wrong

### ISSUE THREAD

**Issue by jrinvictus** [shap/shap#1024]
**Python 3.7
Shap 0.33.0**

I'm seeing different output values compared to the predicted value.  Using the FICO Heloc dataset, for row 13 in the X_test df, the probability should be .45 but the output value in the force_plot is showing .28.  I have reproduced the code below.

Here are some other values that aren't lining up:
i=17
pred = .52
output value = .72

i=20
pred = .15
output value = .01

i=33
pred = .99
output value = .9

```
import pandas as pd #for manipulating data
import numpy as np #for manipulating data
from sklearn.model_selection import train_test_split
import xgboost as xgb
import shap

df=pd.read_csv('heloc_dataset.csv')

target='RiskPerformance'
y=df['RiskPerformance']
df=df.drop(['RiskPerformance'],1)

X_train,X_test,y_train,y_test = train_test_split(df, y, test_size=0.2, random_state=100)

feature_names=X_train.columns

model = xgb.XGBClassifier(max_depth=6,
                    subsample=.5,
                    objective='binary:logistic',
                    n_estimators=50,
                    learning_rate = 1)

model.fit(X_train.values, y_train.values)

explainer = shap.TreeExplainer(model)
expected_value = explainer.expected_value
shap_values = explainer.shap_values(X_test)

i=12
model.predict_proba(X_test.iloc[i])

shap.force_plot(expected_value, shap_values[i], X_test.iloc[i], link='logit', matplotlib=True)
```

**Comment by slundberg** [comment]
Thanks for sharing, this is very strange because passing a single Panda's series does not give the same prediction from XGBoost (0.90) as passing a whole numpy array and then slicing after the fact:
```
> model.predict_proba(X_test.values[:1])
[[0.8875233  0.11247673]]
> model.predict_proba(X_test.iloc[0])
[[0.9214566  0.07854344]]
```

So this seems to be an issue with how XGBoost handles Series objects, and is probably unrelated to SHAP. I have never used Series objects before so I don't know why they seem to produce wrong answers. Perhaps this should be an XGBoost issue?

**Comment by jrinvictus** [comment]
This is great, thanks so much!

If you have time for one quick question, why is expected_value commented out in the gradient.py?

**Comment by slundberg** [comment]
Because self.data could be really large if you are sampling from a full training dataset. So we don't want to force that computation on the user. Perhaps there is a better way, but's that why for now.

**Comment by jrinvictus** [comment]
basically im trying to create decision plots with tensorflow.  Do you have a recommendation on how to do so without the expected value?

**Comment by slundberg** [comment]
Ah. Just subtract the sum of the SHAP values for the  prediction from the
model output. But remember it is a bit of a noisy estimate with
GradientExplainer.

**Comment by jrinvictus** [comment]
Oh yeah that makes way more sense, didn't think about that.  thanks again!
